Detectors/TRD/reconstruction/src/DataReaderTask.cxx

```diff
@@ -62,28 +66,45 @@ void DataReaderTask::run(ProcessingContext& pc)
   auto outputRoutes = pc.services().get<o2::framework::RawDeviceService>().spec().outputs;
   auto fairMQChannel = outputRoutes.at(0).channel;
   int inputcount = 0;
+  std::vector<InputSpec> dummy{InputSpec{"filter", ConcreteDataTypeMatcher{"FLP", "DISTSUBTIMEFRAME"}, Lifetime::Timeframe}};
+  // if we see requested data type input with 0xDEADBEEF subspec and 0 payload this means that the "delayed message"
+  //   // mechanism created it in absence of real data from upstream. Processor should send empty output to not block the workflow
+
+  for (const auto& ref : InputRecordWalker(pc.inputs(), dummy)) {
+    const auto dh = o2::framework::DataRefUtils::getHeader<o2::header::DataHeader*>(ref);
+    if (dh->payloadSize == 16) {
+      LOGP(WARNING, "Found input [{}/{}/{:#x}] TF#{} 1st_orbit:{} Payload {} : assuming no payload for all links in this TF",
+           dh->dataOrigin.str, dh->dataDescription.str, dh->subSpecification, dh->tfCounter, dh->firstTForbit, dh->payloadSize);
+      sendData(pc); //send the empty tf data.
+      return;
+    }
+    LOG(info) << " matched DEADBEEF";
+  }
+  //TODO combine the previous and subsequent loops.
+  int inputcounts = 0;
   /* loop over inputs routes */
   for (auto iit = pc.inputs().begin(), iend = pc.inputs().end(); iit != iend; ++iit) {
+    LOG(info) << " looping over inputs " << inputcounts;
+    inputcounts++;
     if (!iit.isValid()) {
       continue;
     }
     /* loop over input parts */
+    int inputpartscount = 0;
     for (auto const& ref : iit) {
-
+      LOG(info) << " looping over parts " << inputpartscount;
+      if (mVerbose) {
+        const auto dh = DataRefUtils::getHeader<o2::header::DataHeader*>(ref);
+        LOGP(info, "Found input [{}/{}/{:#x}] TF#{} 1st_orbit:{} Payload {} : assuming no payload for all links in this TF",
+             dh->dataOrigin.str, dh->dataDescription.str, dh->subSpecification, dh->tfCounter, dh->firstTForbit, dh->payloadSize);
+      }
```

Comment:
why is this here? Also,  please don't put `LOG(INFO)` in the loop.

Reply:
oops, that was erroneously left in from when i was checking somethings.

Reply:
if you are refering to the LOGP at line 98 its only for debugging via the command line option, there is a hell of a lot of this as mechanism to debug on the flp for when i get real data. They will mostly eventually come out.

Reply:
I was referring to `LOG(info) << " looping over parts " << inputpartscount;`

Reply:
removed